signer/contentsignature/contentsignature_test.go

```diff
@@ -218,3 +219,21 @@ func TestNoShortData(t *testing.T) {
 		t.Fatalf("expected to fail with input data too short but failed with: %v", err)
 	}
 }
+
+// Reproduce nil error
+type ErrorReader struct{}
+
+func (e *ErrorReader) Read(p []byte) (n int, err error) {
+	return 0, errors.New("read error")
+}
+
+func TestRandReadFailureOnSignHash(t *testing.T) {
+	input := []byte("foobarbaz1234abcd")
+	testcase := PASSINGTESTCASES[0]
+	s, _ := New(testcase.cfg)
+	s.rand = &ErrorReader{}
```

Comment:
I think we may want a TODO, at least, to inject `rand.Reader` into either the configuration objects (ugh) or pass them into the `New` constructors here and in contentsignaturepki to allow us to more easily test this code.

I assume y'all took a look at what it would take to pass the Reader into all the `signer/*.New` calls were left hoping for a smaller change.

Reply:
Decided to keep it small and add a TODO. @alexcottner and I wanted to think about it more since the rand not being available is an error case caused by other dependency. Resolved [7b19fee](https://github.com/mozilla-services/autograph/pull/1033/commits/7b19fee2a2e3daac50d64af473f12aa4ea1f72c8)